Generate OpenAPI descriptor

gallotamas opened the pull request:
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | -
| License         | Apache 2.0


### What's in this PR?
Generate OpenAPI descriptor for istio operator


### Why?
External tools (e.g. code editor) might need open api schemas so let's generate it if we already have the toolset for it.


### Additional context
perFile openAPI schema generation doesn't work but it works by generating all types into a single file.
We might want to fix the perFile schema generation (I couldn't make it work so far) if the API becomes too big.

gallotamas committed:
feat(api): generate OpenAPI descriptor

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/banzaicloud/istio-operator?pullRequest=805) <br/>All committers have signed the CLA.

gallotamas committed:
Merge branch 'release-1.11' into feat/generate-openapi

waynz0r committed:
cue gen enhancements
- move field options to a different package
- support special intorstring and quantity types
- bump cue-gen version

waynz0r committed:
generate code

Laci21 commented on api/options/options.proto:
```diff
@@ -0,0 +1,29 @@
+
+package istio_operator.v2.api.options;
+
+option go_package = "github.com/banzaicloud/istio-operator/api/v2/options;options";
```
just checking: `options;options` is intended to be like this?

waynz0r replied:
it was intended, but `;options` is not needed, so it got removed.

waynz0r committed:
fix options go package name

waynz0r committed:
Merge branch 'release-1.11' into feat/generate-openapi
